src/hypofuzz/provider.py

```diff
@@ -136,3 +570,61 @@ def draw_bytes(
         choice = self._pop_choice("bytes", {"min_size": min_size, "max_size": max_size})
         assert isinstance(choice, bytes)
         return choice
+
+
+@lru_cache
```

Comment:
let's unbound this cache size, and worker_identity, because there could be quite a few directories and the dicts are cheap.